jsettlers.main.swing/src/main/java/jsettlers/main/swing/lookandfeel/ui/ButtonUiStone.java

```diff
@@ -15,138 +15,161 @@
 package jsettlers.main.swing.lookandfeel.ui;
 
 import java.awt.Color;
+import java.awt.Dimension;
 import java.awt.FontMetrics;
 import java.awt.Graphics;
-import java.awt.Rectangle;
+import java.awt.Graphics2D;
 import java.awt.image.BufferedImage;
 
 import javax.swing.AbstractButton;
 import javax.swing.ButtonModel;
 import javax.swing.JComponent;
 import javax.swing.JToggleButton;
-import javax.swing.border.Border;
-import javax.swing.plaf.basic.BasicToggleButtonUI;
+import javax.swing.plaf.basic.BasicButtonUI;
 
+import jsettlers.main.swing.lookandfeel.DrawHelper;
 import jsettlers.main.swing.lookandfeel.ui.img.UiImageLoader;
-import sun.swing.SwingUtilities2;
 
 /**
- * Stone toggle Button UI
- * 
+ * Button UI Implementation
+ *
  * @author Andreas Butti
  */
-public class ButtonUiStone extends BasicToggleButtonUI {
-
-	private static final String CLIENT_PROPERTY_RANDOM_BACKGROUND_X_OFFSET = "ButtonUiStone.randomBackgroundXOffset";
-	private static final String CLIENT_PROPERTY_RANDOM_BACKGROUND_Y_OFFSET = "ButtonUiStone.randomBackgroundYOffset";
+public class ButtonUiStone extends BasicButtonUI {
 
 	/**
 	 * Background Image
 	 */
-	private final BufferedImage backgroundImage = UiImageLoader.get("granit_texture1.png");
+	private final BufferedImage backgroundImage = UiImageLoader.get("sr_ui_button/sr_ui_button-bg.png");
 
 	/**
-	 * Button down
+	 * Background Image pressed
 	 */
-	private final Border borderDown;
+	private final BufferedImage backgroundImagePressed = UiImageLoader.get("sr_ui_button_down/sr_ui_button-bg.png");
 
 	/**
-	 * Button up
+	 * Border images if the Button is not pressed
 	 */
-	private final Border borderUp;
+	private final BufferedImage[] BORDER_NORMAL = { UiImageLoader.get("sr_ui_button/sr_ui_button-corner-upper-left.png"),
+			UiImageLoader.get("sr_ui_button/sr_ui_button-border-top.png"),
+			UiImageLoader.get("sr_ui_button/sr_ui_button-corner-upper-right.png"),
+			UiImageLoader.get("sr_ui_button/sr_ui_button-border-right.png"),
+			UiImageLoader.get("sr_ui_button/sr_ui_button-corner-bottom-right.png"),
+			UiImageLoader.get("sr_ui_button/sr_ui_button-border-bottom.png"),
+			UiImageLoader.get("sr_ui_button/sr_ui_button-corner-bottom_left.png"),
+			UiImageLoader.get("sr_ui_button/sr_ui_button-border-left.png")
+	};
 
 	/**
-	 * Offset
+	 * Border images if the Button is pressed
 	 */
-	private int shiftOffset = 0;
+	private final BufferedImage[] BORDER_DOWN = { UiImageLoader.get("sr_ui_button_down/sr_ui_button-corner-upper-left.png"),
+			UiImageLoader.get("sr_ui_button_down/sr_ui_button-border-top.png"),
```

Comment:
what does the sr_ in the folder names stand for?

Reply:
You have to ask Ben, I got the graphics with this names

Reply:
Localize the header of the Dialog is no Problem. But this seems to be the Exception handler.
So to localize all messages, you have to localize all Exceptions like this: https://stackoverflow.com/questions/24988491/difference-between-e-getmessage-and-e-getlocalizedmessage

I generally prefer the way to handle the expected message, and provide a userfriendly string for common messages.

In all other cases, I display usually a general like, like "an error occurred, contact the developer bla bla", the error is: bla bla

Standard exceptions are never translated on windows with Oracle JDK, but always translated on Linux with OpenJDK (Connection Lost, File not found etc.)

@andreas-eberle What do you think?

Reply:
@andreasb242: I think the map loading error is an error we have to expect (when you switch to a new version of JSettlers, you will probably end up with this error when you load a savegame / you might get it from original maps.) 

Therefore I think we should localize this error. I guess it should be possible to catch MapLoadingException and then display a special dialog heading and conent and add the message of the exception for debugging reasons. I think that would be kind of a middle way. What do you think?